src/shared/install.c

```diff
@@ -1893,7 +1893,7 @@ int unit_file_unmask(
                 if (!GREEDY_REALLOC0(todo, n_allocated, n_todo + 2))
                         return -ENOMEM;
 
-                todo[n_todo++] = *i;
+                todo[n_todo++] = strdup(*i);
```

Comment:
Fixed in  https://github.com/systemd/systemd/pull/5054/commits/d054eae6c954baa857170bb60072c8a2ecea0d6b